src/static/riot/analytics/analytics.tag

```diff
@@ -0,0 +1,520 @@
+                            }
+                        }],
+                        yAxes: [{
+                            ticks: {
+                                beginAtZero: true
+                            }
+                        }]
+                    }
+                }
+            }
+        }
+
+        function build_chart_data(data, day_resolution, csv_format) {
+            var chart_data = []
+            var x_data = []
+            var y_data = []
+
+            for (var year in data) {
+                if(!data.hasOwnProperty(year)) continue;
+
+                var year_obj = data[year]
+                for(var month in year_obj) {
+                    if(!year_obj.hasOwnProperty(month)) continue;
+
+                    if(day_resolution) {
+                        var month_obj = data[year][month]
+                        for(var day in month_obj) {
+                            if(!month_obj.hasOwnProperty(day)) continue;
+
+                            if (year_obj[month][day].total === undefined) {
+                            }
+                            if (csv_format) {
+                                x_data.push(new Date(year, month - 1, day).toJSON().slice(0,10))
+                                y_data.push(year_obj[month][day].total)
+                            } else {
+                                chart_data.push({
+                                    x: new Date(year, month - 1, day),
+                                    y: year_obj[month][day].total
+                                })
+                            }
+                        }
+                    } else {
+                        if (csv_format) {
+                                x_data.push(new Date(year, month - 1).toJSON().slice(0,10))
+                                y_data.push(year_obj[month].total)
+                        } else {
+                            chart_data.push({
+                                x: new Date(year, month - 1),
+                                y: year_obj[month].total
+                            })
+                        }
+                    }
+                }
+            }
+            if (csv_format) {
+                let d = {
+                    x: x_data,
+                    y: y_data,
+                }
+                return d
+            } else {
+                return chart_data
+            }
+         }
+
+        function update_chart(chart, data, day_resolution) {
+            chart.data.datasets[0].data = build_chart_data(data, day_resolution, false)
+            chart.update()
+        }
+
+        self.update_analytics = function (start, end, time_unit) {
+
+            var start_date = new Date(start.year, start.month, start.day)
+            if (end) {
+                var end_date = new Date(end.year, end.month, end.day)
+            } else {
+                var end_date = new Date()
+                end_date.setDate(end_date.getDate() + 1)
```

Comment:
setting the end date as tomorrow you could do:
```javascript
var end_date = new Date(_.now() + 86400000)  // the number of milliseconds in 24 hours i.e. 24 * 60 * 60 * 1000
```

Reply:
Nice. More concise.